app/src/features/sources/SourceCreate.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+
+import { Button, makeStyles, Typography } from "@material-ui/core";
+import { useTranslation } from "react-i18next";
+
+import { useAppDispatch } from "../../app/store";
+import { Source } from "../../services/api/generated/api.generated";
+import { editSource } from "./sourceSlice";
+
+const useStyles = makeStyles(() => ({
+  button: {
+    textTransform: "none",
+  },
+}));
+
+const SourceCreate = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const dispatch = useAppDispatch();
+
+  const handleClick = () => dispatch(editSource({ name: "" } as Source));
```

Comment:
so this is the trick to open the drawer to create a source ? I yes, I think it deserves a comment to say so

Reply:
Here we're merely saying we want to edit this new initialized Source. It's up to SourceForm to tell if it wants to open a drawer to edit a Source. I will add a comment.